base/workerpool.jl

```diff
@@ -114,3 +117,57 @@ using `remotecall_fetch`.
 """
 remote(f) = (args...; kwargs...)->remotecall_fetch(f, default_worker_pool(), args...; kwargs...)
 remote(p::WorkerPool, f) = (args...; kwargs...)->remotecall_fetch(f, p, args...; kwargs...)
+
+type CachedWorkerPool # Manages a cache for code/data on workers.
+    pool::WorkerPool
+    directory::Dict{Tuple, RemoteChannel}  # Mapping between a tuple (worker_id, object_id) and a remote_ref
+    CachedWorkerPool(pool) = new(pool, Dict{Int, Any}())
+end
+
+function release(p::CachedWorkerPool)  # For a quick release of remote function cache
```

Comment:
What are the pro's and cons of `release` being a method
vs registering this as the finalizer for `CachedWorkerPool`,
and then if you want to release it early, you can call `finalize` on the CachedWorkerPool.
After release is called the CachedWorkerPool is already trashed and might as well be garbage collected, since all information it was meant to hold was finalized and referenced deleted.

This also might allow for the release calls to be removed from  the pmap call?
Since the finalizer would just be called when it goes out of scope anyway.